geomesa-nifi-processors/geomesa-nifi-processors-accumulo/src/test/scala/org/geomesa/nifi/processors/accumulo/PutGeoMesaAccumuloTest.scala

```diff
@@ -0,0 +1,132 @@
+      runner.assertTransferCount(AbstractGeoIngestProcessor.Relationships.FailureRelationship, 0)
+    } finally {
+      runner.shutdown()
+    }
+
+    val ds = DataStoreFinder.getDataStore((dsParams + (AccumuloDataStoreParams.CatalogParam.key -> catalog)).asJava)
+    Assert.assertNotNull(ds)
+    try {
+      val sft = ds.getSchema("example")
+      Assert.assertNotNull(sft)
+      val features = SelfClosingIterator(ds.getFeatureSource("example").getFeatures.features()).toList
+      logger.debug(features.mkString(";"))
+      Assert.assertEquals(3, features.length)
+    } finally {
+      ds.dispose()
+    }
+  }
+
+  @Test
+  def testIngestAttributes(): Unit = {
+    val catalog = s"${root}IngestAttributes"
+    val runner = TestRunners.newTestRunner(new PutGeoMesaAccumulo())
+    try {
+      dsParams.foreach { case (k, v) => runner.setProperty(k, v) }
+      runner.setProperty(AccumuloDataStoreParams.CatalogParam.key, catalog)
+      runner.setProperty(AbstractGeoIngestProcessor.Properties.IngestModeProp, IngestMode.Converter)
+      val attributes = new java.util.HashMap[String, String]()
+      attributes.put(AbstractGeoIngestProcessor.Attributes.SftSpecAttribute, "example")
+      attributes.put(AbstractGeoIngestProcessor.Attributes.ConverterAttribute, "example-csv")
+      runner.enqueue(getClass.getClassLoader.getResourceAsStream("example.csv"), attributes)
+      runner.run()
+      runner.assertTransferCount(AbstractGeoIngestProcessor.Relationships.SuccessRelationship, 1)
+      runner.assertTransferCount(AbstractGeoIngestProcessor.Relationships.FailureRelationship, 0)
+    } finally {
+      runner.shutdown()
+    }
+
+    val ds = DataStoreFinder.getDataStore((dsParams + (AccumuloDataStoreParams.CatalogParam.key -> catalog)).asJava)
+    Assert.assertNotNull(ds)
+    try {
+      val sft = ds.getSchema("example")
+      Assert.assertNotNull(sft)
+      val features = SelfClosingIterator(ds.getFeatureSource("example").getFeatures.features()).toList
+      logger.debug(features.mkString(";"))
+      Assert.assertEquals(3, features.length)
+    } finally {
+      ds.dispose()
+    }
+  }
+
+  @Test
+  def testIngestAttributeOverride(): Unit = {
+    val catalog = s"${root}IngestAttributeOverride"
+    val runner = TestRunners.newTestRunner(new PutGeoMesaAccumulo())
+    try {
+      dsParams.foreach { case (k, v) => runner.setProperty(k, v) }
+      runner.setProperty(AccumuloDataStoreParams.CatalogParam.key, catalog)
+      runner.setProperty(AbstractGeoIngestProcessor.Properties.IngestModeProp, IngestMode.Converter)
+      runner.setProperty(AbstractGeoIngestProcessor.Properties.SftNameKey, "example")
+      runner.setProperty(AbstractGeoIngestProcessor.Properties.ConverterNameKey, "example-csv")
+      val attributes = new java.util.HashMap[String, String]()
+      attributes.put(AbstractGeoIngestProcessor.Attributes.SftNameAttribute, "renamed")
```

Comment:
Ok, I'm glad I read this unit test...  let me see if I've got this straight...

In this example, no SFT (and hence no tables) named 'renamed' existed before this FlowFile was enqueued?  But this file being ingested would have created the SFT and tables?

This seems like a Very Dangerous Feature.  One misconfiguration, and a NiFi could call for the creation of countless tables in a distributed database?  😬

Reply:
sounds about right :)

Reply:
Do we have the option to *not* do it that way?  I'd hate to shred someone's database in that manner...

Reply:
well, that's the whole point of this PR...

Reply:
Is there a way to add some kind of reasonable limit here?  E.g., you can select any of the existing SFTs which are on the classpath?

Maybe the 'full' solution for what I'm wanting would be to use the Avro Schema Registry as a brokerage for SFT names?